### PR TITLE
TKNECO-93: Buildah Task Documentation

### PR DESCRIPTION
**Pull request by Senjuti256**
Basic documentation as per story: https://issues.redhat.com/browse/TKNECO-92

Drafted a preliminary documentation for users to get a hold of the Buildah Task.

### CODE REVIEW

File: doc/skopeo-copy.md
Comment (otaviof) [inline doc/skopeo-copy.md:1]: Please sync up with @Aneesh-M-Bhat's PR #40, check if there are any differences from what you're doing with his and merge into a single place.

File: doc/buildah.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+Containers Tekton Tasks
+-----------------------
+
+# `Buildah` Tekton Task
```
Comment (otaviof) [inline doc/buildah.md:4]: Please the following snippet for the page tittle:

```
Buildah Tekton Task
-------------------
```

File: doc/buildah.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+In case the Container Registry requires authentication, please consider the [Tekton Pipelines documentation][tektonPipelineAuth]. In a nutshell, you need to create a Kubernetes Secret describing the following attributes:
+
+```bash
+kubectl create secret docker-registry imagestreams \
+  --docker-server="image-registry.openshift-image-registry.svc:5000" \
+  --docker-username="${REGISTRY_USERNAME}" \
+  --docker-password="${REGISTRY_TOKEN}"
+```
+
+Then make sure the Secret is linked with the Service-Account running the `TaskRun`/`PipelineRun`.
```
Comment (otaviof) [inline doc/buildah.md:41]: On the previous PR (#40) we end up repeating this part, let's please extract the container registry authentication configuration into a single `.md` file, so we can link this and future documentation to the same place.

File: doc/buildah.md
Comment (otaviof) [inline doc/buildah.md:1]: Let's use a `docs` folder, please.

File: doc/buildah.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+spec:
+  pipelineRef:
+    name: task-buildah
+  params:
+    - name: IMAGE
+      value: your-image-name
+    - name: TLS_VERIFY
+      value: true
+    - name: VERBOSE
+      value: false
```
Comment (otaviof) [inline doc/buildah.md:25]: For a shorter example, you can omit the Params with default values, like these two.

File: doc/buildah.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+| Param         | Type                       | Default                      | Description                |
+| :------------ | :------------------------: | :--------------------------- | :------------------------- |
+| `IMAGE` | `string` | (required) | Fully qualified source container image name, including tag, to be built by buildah. |
+| `CONTAINERFILE_PATH` | `string` | `Containerfile` | Path to the `Containerfile` (or `Dockerfile`) relative to the `source` workspace. |
+| `TLS_VERIFY` | `string` | `true` | Sets the TLS verification flags, `true` is recommended. |
+| `VERBOSE` | `string` | `false` | Shows a more verbose (debug) output. |
+| `SUBDIRECTORY` | `string` | `.` | Relative subdirectory to the `source` Workspace for the build-context. |
+| `STORAGE_DRIVER` | `string` | `overlay` | Set buildah storage driver to reflect the currrent cluster node's settings. |
+| `BUILD_EXTRA_ARGS` | `string` | `""` | Extra parameters passed for the build command when building images. |
+| `PUSH_EXTRA_ARGS` | `string` | `""` | Extra parameters passed for the push command when pushing images. |
```
Comment (otaviof) [inline doc/buildah.md:54]: Empty should just be ``, instead of `""`.

File: doc/buildah.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+Containers Tekton Tasks
+-----------------------
+
+# `Buildah` Tekton Task
+
+The `buildah` Task is meant to build [OCI][OCI] container images without the requirement of container runtime daemon like Docker daemon using [Buildah][containersBuildah], the Task results contain the image name and the SHA256 image digest.
```
Comment (otaviof) [inline doc/buildah.md:6]: Please extract this paragraph on a `# Abstract` document section.

File: doc/buildah.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+Please, consider the usage example below:
```
Comment (otaviof) [inline doc/buildah.md:8]: And let's create a new section named `# Usage`, please.

File: doc/buildah.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+```bash
+kubectl create secret docker-registry imagestreams \
+  --docker-server="image-registry.openshift-image-registry.svc:5000" \
+  --docker-username="${REGISTRY_USERNAME}" \
+  --docker-password="${REGISTRY_TOKEN}"
+```
+
+Then make sure the Secret is linked with the Service-Account running the `TaskRun`/`PipelineRun`.
+
+## Params
```
Comment (otaviof) [inline doc/buildah.md:43]: Please add a new section for the Workspaces documentation.

**Review by otaviof — CHANGES_REQUESTED** [review]
Good write up, thank you! Please, align the modifications following #40, and consider the review comments as well.

File: doc/buildah.md
Comment (Senjuti256) [reply doc/buildah.md:1]: I have used it already.

File: doc/skopeo-copy.md
Comment (Senjuti256) [reply doc/skopeo-copy.md:1]: Sure

**Pull request closed without merging**